finality-aleph/src/network/tests.rs

```diff
@@ -0,0 +1,423 @@
+    {
+        assert_eq!(message, bytes);
+        assert_eq!(peer_id_bytes, data.bob.peer_id.to_bytes());
+    } else {
+        panic!("expected message received network event")
+    }
+    data.complete().await;
+}
+#[tokio::test]
+async fn test_network_commands() {
+    let mut data = prepare_one_session_test_data().await;
+    data.network.emit_event(Event::NotificationStreamOpened {
+        remote: data.bob.peer_id,
+        protocol: PROTOCOL_NAME.into(),
+        role: ObservedRole::Authority,
+    });
+    data.network.emit_event(Event::NotificationStreamOpened {
+        remote: data.charlie.peer_id,
+        protocol: PROTOCOL_NAME.into(),
+        role: ObservedRole::Authority,
+    });
+
+    println!("send to peer");
+
+    // SendToPeer
+    {
+        let fake_message: Vec<u8> = vec![157];
+        data.rush_network
+            .net_command_tx
+            .send(SessionCommand {
+                session_id: data.rush_network.session_id,
+                command: NetworkCommand::SendToPeer(
+                    fake_message.clone(),
+                    data.bob.peer_id.to_bytes(),
+                ),
+            })
+            .await
+            .unwrap();
+        match data.network.send_message.1.lock().next().await {
+            Some((peer_id, protocol, message)) => {
+                assert_eq!(peer_id, data.bob.peer_id);
+                assert_eq!(protocol, PROTOCOL_NAME);
+                match <(SessionId, Vec<u8>) as Decode>::decode(&mut message.as_slice()) {
+                    Ok((session_id_, message)) => {
+                        assert_eq!(session_id_.0, 0);
+                        assert_eq!(message, fake_message);
+                    }
+                    _ => panic!("Expected a message"),
+                }
+            }
+            _ => panic!("Expecting a network message"),
+        }
+    }
+
+    println!("send to all");
+
+    // SendToAll
+    {
+        let fake_message: Vec<u8> = vec![205];
+        data.rush_network
+            .net_command_tx
+            .send(SessionCommand {
+                session_id: data.rush_network.session_id,
+                command: NetworkCommand::SendToAll(fake_message.clone()),
+            })
+            .await
+            .unwrap();
+        let mut peer_ids = HashSet::<PeerId>::new();
+        for _ in 0..2_u8 {
+            match data.network.send_message.1.lock().next().await {
+                Some((peer_id, protocol, message)) => {
+                    peer_ids.insert(peer_id);
+                    assert_eq!(protocol, PROTOCOL_NAME);
+                    match <(SessionId, Vec<u8>)>::decode(&mut message.as_slice()) {
+                        Ok((session_id, message)) => {
+                            assert_eq!(session_id.0, 0);
+                            assert_eq!(message, fake_message);
+                        }
+                        _ => panic!("Expected a request parents consensus message"),
```

Comment:
This panic message does not sound right, probably some copy-paste issue. Happens also below couple of times.

Reply:
Oops... fixed that